sparkly/utils.py

```diff
@@ -59,6 +64,89 @@ def absolute_path(file_path, *rel_path):
     )
 
 
+def chunks(seq, n):
+    """Iterate over seq by chunks of n elements.
+
+    >>> list(chunks([1, 2, 3, 4, 5], 2))
+    [[1, 2], [3, 4], [5]]
+    >>> list(chunks([], 2))
+    []
+    >>> list(chunks([1], 2))
+    [[1]]
+    """
+    it = iter(seq)
+    while True:
+        chunk = list(islice(it, n))
+        if not chunk:
+            break
+        yield chunk
+
+
+def type_has(t, required_fields):
+    """Check whether a complex dataType has specific fields.
+
+    Args:
+        t (pyspark.sql.types.ArrayType, pyspark.sql.types.MapType or
```

Comment:
don't have a use-case, but the signature will be much simpler in this case:
```
t (DataType) instead of t(ArrayType, MapType, StructType)
```
I am ok to leave it as is.